files/hello_world_example_app/app/src/main/cpp/main.cpp

```diff
@@ -0,0 +1,104 @@
+#include <stdarg.h>
+#include <stdio.h>
+#include <sstream>
+#include <map>
+#include <string.h>
+#include <errno.h>
+#include <vector>
+#include <set>
+#include <fstream>
+#include <android/log.h>
+
+#include "ros/ros.h"
+#include <std_msgs/String.h>
+
+#include <android_native_app_glue.h>
+
+int loop_count_ = 0;
+ros::Publisher chatter_pub;
+
+void log(const char *msg, ...) {
+    va_list args;
+    va_start(args, msg);
+    __android_log_vprint(ANDROID_LOG_INFO, "ROSCPP_NDK_EXAMPLE", msg, args);
+    va_end(args);
+}
```

Comment:
Why not use the normale rosconsole logging macros with `-DROSCONSOLE_BACKEND=android`, the default since #53?

Reply:
Yes, I will modify it to use rosconsole